File: src/main/java/guru/xgm/angelscript/lang/annotation/ZincAnnotator.java
package guru.xgm.angelscript.lang.annotation;

import com.intellij.lang.annotation.AnnotationHolder;
import com.intellij.lang.annotation.Annotator;
import com.intellij.lang.annotation.HighlightSeverity;
import com.intellij.psi.PsiElement;
import guru.xgm.angelscript.openapi.fileTypes.ZincSyntaxHighlighterBase;
import guru.xgm.angelscript.psi.ZincTypeName;
import org.jetbrains.annotations.NotNull;

final class ZincAnnotator implements Annotator {
    @Override
    public void annotate(@NotNull final PsiElement element, @NotNull AnnotationHolder holder) {

        if (element instanceof ZincTypeName type) {
            holder
                    .newSilentAnnotation(HighlightSeverity.INFORMATION)
                    .range(type.getTextRange())
                    .textAttributes(ZincSyntaxHighlighterBase.TYPE_NAME_KEY).create();
        }
    }
}

File: src/main/kotlin/raft/war/language/zinc/lang/annotation/ZincAnnotator.kt
package raft.war.language.zinc.lang.annotation

import com.intellij.lang.annotation.AnnotationHolder
import com.intellij.lang.annotation.Annotator
import com.intellij.lang.annotation.HighlightSeverity
import com.intellij.psi.PsiElement
import raft.war.language.zinc.openapi.fileTypes.ZincSyntaxHighlighterBase
import raft.war.language.zinc.psi.ZincTypeName

internal class ZincAnnotator : Annotator {
    override fun annotate(element: PsiElement, holder: AnnotationHolder) {
        if (element is ZincTypeName) {
            holder
                .newSilentAnnotation(HighlightSeverity.INFORMATION)
                .range(element.getTextRange())
                .textAttributes(ZincSyntaxHighlighterBase.ZINC_TYPE_NAME).create()
        }
    }
}
